test/c/test_target.c

```diff
@@ -544,6 +545,98 @@ static int test_target_update_instruction(void) {
     return result;
 }
 
+/**
+ * Test if our target is compatible with certain instructions.
+ */
+int test_target_instruction_supported(void) {
+    QkTarget *sample_target = create_sample_target(true);
+    int result = Ok;
+
+    char *gate_names[5] = {"x", "y", "id", "rz", "sx"};
+    QkParam **params[5] = {NULL, NULL, NULL,
+                           (QkParam *[]){
+                               qk_param_from_double(3.14),
+                           },
+                           NULL};
```

Comment:
A minor style suggestion (alternatively, I miss a ternary operator in Rust and here is an opportunity to use one 😉 ): 
```c
QkParam *param[1] = {qk_param_from_double(3.14)};
```
Followed by: 
```c
 if (!(qk_target_instruction_supported(sample_target, gate_names[j], qargs, j != 3 ? NULL : params)
```